crates/weaver_forge/src/extensions/otel.rs

```diff
@@ -230,23 +257,146 @@ pub(crate) fn is_deprecated(input: Value) -> bool {
     false
 }
 
+/// Returns the instantiated type of the input type.
+pub(crate) fn instantiated_type(attr_type: &Value) -> Result<String, minijinja::Error> {
+    if is_simple_type(attr_type) {
+        return Ok(attr_type
+            .as_str()
+            .expect("should never happen, already tested in is_template_type")
+            .to_owned());
+    }
+    if is_template_type(attr_type) {
+        let attr_type = attr_type
+            .as_str()
+            .expect("should never happen, already tested in is_template_type");
+        let end = attr_type.len() - TEMPLATE_SUFFIX.len();
+        return Ok(attr_type[TEMPLATE_PREFIX.len()..end].to_owned());
+    }
+    if is_enum_type(attr_type) {
+        return enum_type(attr_type);
+    }
+    Err(minijinja::Error::custom(format!(
+        "Expected simple type, template type, or enum type, found {}",
+        attr_type
+    )))
+}
+
+/// Returns the inferred enum type of the input type or an error if the input type is not an enum.
+pub(crate) fn enum_type(attr_type: &Value) -> Result<String, minijinja::Error> {
+    if let Ok(members) = attr_type.get_attr("members") {
+        // Infer the enum type from the members.
+        let mut inferred_type: Option<String> = None;
+        for member in members.try_iter()? {
+            let value = member.get_attr("value")?;
+            let member_type = match value.kind() {
+                ValueKind::Number => {
+                    if value.as_i64().is_some() {
+                        "int"
+                    } else {
+                        "double"
+                    }
+                }
+                ValueKind::String => "string",
+                _ => {
+                    return Err(minijinja::Error::custom(format!(
+                        "Enum values are expected to be int, double, or string, found {}",
+                        value
+                    )));
+                }
+            };
+            inferred_type = match inferred_type {
+                Some(current_inferred_type) => {
+                    if current_inferred_type != member_type {
+                        // If the inferred type is different from the member type, then the enum
+                        // type is "promoted" to a string.
+                        Some("string".to_owned())
```

Comment:
I believe this is the right way to go... We may need to improve enum in the model so type is actually specified AND enforced on members.